error.c

```diff
@@ -3860,6 +3860,12 @@ void
 rb_error_frozen_object(VALUE frozen_obj)
 {
     rb_yjit_lazy_push_frame(GET_EC()->cfp->pc);
+
+    if (CHILLED_STRING_P(frozen_obj)) {
+        CHILLED_STRING_MUTATED(frozen_obj);
```

Comment:
I first thought it was strange that this was done here (and after `rb_yjit_lazy_push_frame`), and not as part of the implementation of `rb_check_frozen_inline` (which calls this).  _however_, I see it really actually is required. The documentation for `rb_yjit_lazy_push_frame` says:

> Push a C method frame if the given PC is supposed to lazily push one.
> This is called from rb_raise() (at rb_exc_new_str()) and other functions
> that may make a method call (e.g. rb_to_int()).

And CHILLED_STRING_MUTATED _can_ wind up making a method call because warnings are implemented by actually calling `::Warning.warn`.

I guess this isn't news to anybody but me though :P